internal/session.go

```diff
@@ -195,7 +196,7 @@ func CreateSession(ctx Context, sessionOptions *SessionOptions) (Context, error)
 // The main usage of RecreateSession is for long sessions that are splited into multiple runs. At the end of
 // one run, complete the current session, get recreateToken from sessionInfo by calling SessionInfo.GetRecreateToken()
 // and pass the token to the next run. In the new run, session can be recreated using that token.
-func RecreateSession(ctx Context, recreateToken []byte, sessionOptions *SessionOptions) (Context, error) {
+func RecreateSession(ctx Context, recreateToken *commonpb.Payload, sessionOptions *SessionOptions) (Context, error) {
```

Comment:
Token should stay as a binary payload. It is an opaque blob from the client point of view.

Reply:
`deserializeRecreateToken` at the line bellow calls `dataConverter`. It must be `Payload` or data converter should not be used (just old plain JSON instead).